### ISSUE TITLE
babel-plugin-transform-template-literals literals get overwritten if transpiled in multiple script tags

### ISSUE THREAD

**Issue by daKmoR** [babel/babel#7321]
> Choose one: is this a bug report or feature request?

Bug Report

TLDR: two different transpiled script tags having template literals - should have h1 and h3... but only the last template literal wins.
Example: https://codepen.io/daKmoR/pen/LQGKRX

### Input Code
Two Script tags which get's transpiled to this
```js
<textarea style="width: 400px; height: 400px;" id="mytop"></textarea>
<textarea style="width: 400px; height: 400px;" id="mybottom"></textarea>

<script>
var _templateObject = _taggedTemplateLiteral(["<h1>", "</h1>"], ["<h1>", "</h1>"]);

function _taggedTemplateLiteral(strings, raw) { return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }

var process = function process(text, data) {
  return "Template Literal: \n" + JSON.stringify(text.raw, null, 2) + "\n\nData: \n" + JSON.stringify(data, null, 3);
};

var name = 'foo';

setTimeout(function () {
  var result = process(_templateObject, name);
  var target = document.querySelector('#mytop');
  target.innerHTML = result;
}, 0);
</script>

<script>
var _templateObject = _taggedTemplateLiteral(["<h3>", "</h3>"], ["<h3>", "</h3>"]);

function _taggedTemplateLiteral(strings, raw) { return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }

var process = function process(text, data) {
  return "Template Literal: \n" + JSON.stringify(text.raw, null, 2) + "\n\nData: \n" + JSON.stringify(data, null, 3);
};

var name = 'foo';

setTimeout(function () {
  var result = process(_templateObject, name);
  var target = document.querySelector('#mybottom');
  target.innerHTML = result;
}, 0);
</script>
```

heaving a "global" static name `_templateObject` seems dangerous
it works fine if you run it in a single script tag... but what if I have multiple script tags?

Script Tag 1: https://codepen.io/daKmoR/pen/ddGELq
Script Tag 2: https://codepen.io/daKmoR/pen/zRrVvp
Result in Failure: https://codepen.io/daKmoR/pen/LQGKRX

### Expected Behavior
- to not have on script tag overwrite template literals from another script tag

### Current Behavior
- last Template Literal overwrites all before

### Possible Solution
- unique name for _templateObject or a wrap it in a closure
- or if there is already a closure don't put your code outside of it

### Context
Because of this I can't use template literals... as I have multiple script tags on one page.

| software         | version(s)
| ---------------- | -------
| Babel            |  
| Babylon          | 
| node             | 
| npm              | 
| Operating System | 

Software Versions are not needed as errors can be reproduced on codepen.

**Comment by daKmoR** [comment]
Additional Info:

Even if there is already a closure
```js
(() => { let process = function(text, data) {
  return "Template Literal: \n" + JSON.stringify(text.raw, null, 2) + "\n\nData: \n" + JSON.stringify(data, null, 3);
}

let name = 'foo';

setTimeout(() => { 
  let result = process`<h1>${name}</h1>`;
  let target = document.querySelector('#mytop');
  target.innerHTML = result;
}, 0);
})();
```

the relevant code will be put outside
```js
var _templateObject = _taggedTemplateLiteral(["<h1>", "</h1>"], ["<h1>", "</h1>"]);

function _taggedTemplateLiteral(strings, raw) { return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }

(function () {
  var process = function process(text, data) {
    return "Template Literal: \n" + JSON.stringify(text.raw, null, 2) + "\n\nData: \n" + JSON.stringify(data, null, 3);
  };

  var name = 'foo';

  setTimeout(function () {
    var result = process(_templateObject, name);
    var target = document.querySelector('#mytop');
    target.innerHTML = result;
  }, 0);
})();
```

if a closure would solve the problem I would surely go for it.

**Comment by LarsDenBakker** [comment]
+1 ran into the same issue as well

**Comment by loganfsmyth** [comment]
Generally Babel makes the assumption that individual files have their own scope. If that is not the case, I'd recommend using something like https://github.com/TrySound/babel-plugin-iife-wrap to wrap your code in an IIFE automatically. That plugins wraps it late enough that it should work.
